src/data/finset/n_ary.lean

```diff
@@ -333,4 +334,29 @@ lemma image₂_right_identity {f : γ → β → γ} {b : β} (h : ∀ a, f a b
   image₂ f s {b} = s :=
 by rw [image₂_singleton_right, funext h, image_id']
 
+variables [decidable_eq α] [decidable_eq β]
+
+lemma image₂_inter_union_subset {f : α → α → β} {s t : finset α} (hf : ∀ a b, f a b = f b a) :
+  image₂ f (s ∩ t) (s ∪ t) ⊆ image₂ f s t :=
+coe_subset.1 $ by { push_cast, exact image2_inter_union_subset hf }
+
+lemma image₂_union_inter_subset {f : α → α → β} {s t : finset α} (hf : ∀ a b, f a b = f b a) :
+  image₂ f (s ∪ t) (s ∩ t) ⊆ image₂ f s t :=
+coe_subset.1 $ by { push_cast, exact image2_union_inter_subset hf }
+
 end finset
+
+namespace set
+variables [decidable_eq γ] {s : set α} {t : set β}
+
+@[simp] lemma to_finset_image2 (f : α → β → γ) (s : set α) (t : set β) [fintype s] [fintype t]
```

Comment:
Why this is not `image2_to_finset`?

Reply:
The outer function, `to_finset` comes first. This matches [`set.to_finset_image`](https://leanprover-community.github.io/mathlib_docs/data/fintype/basic.html#set.to_finset_image)